packages/date-picker/test/keyboard-input.test.js

```diff
@@ -502,6 +502,115 @@ describe('keyboard', () => {
       expect(result.getFullYear()).to.equal(2001);
     });
 
+    it('should parse three digits with short year 4', async () => {
```

Comment:
Nit: Tests could be improved. This series of tests is specifically for using the current date as reference date, which could be reflected in the test case names. Apart from that individual test cases don't have a meaningful name.

The cases could be reduced to just two test cases:
- `it should parse short year with current date as reference date`
- `it should parse short year with custom reference date`

Both could just run through all cases (currently labeled as 1-8) using a utility function that receives:
- the offset that should be added to the entered year
- the offset that should be added to the expected year

So something like:
```
it('should parse short year with current date as reference date', () => {
  checkYearOffset(-49, 0);
  checkYearOffset(49, 0);
  checkYearOffset(-51, 100);
  checkYearOffset(51, -100);
  ...
});
```

Note that the existing three test cases are theoretically obsolete, but will only fail in a few decades 🙂. I think they can be removed.